app/server/utils/metrics_cache.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime, timedelta
+
+from server import red, db
+from flask import g
+from decimal import Decimal
+import json
+import pickle
+
+SUM = 'SUM'
+SUM_OBJECTS ='SUM_OBJECTS'
+TIMESERIES ='TIMESERIES'
+COUNT ='COUNT'
+
+valid_strategies = [SUM, TIMESERIES, COUNT, SUM_OBJECTS]
+
+def _store_cache(key, value):
+    pickled_object = pickle.dumps(value)
+    red.set(key, pickled_object)
+    return True
+
+def _load_cache(key):
+    cached_object = red.get(key)
+    if not cached_object:
+        return None
+    return pickle.loads(cached_object)       
+
+def execute_with_partial_history_cache(metric_name, query, object_model, strategy, disable_cache = False):
+    # disable_cache pass-thru. This is so we don't cache data when filters are active.
+    if disable_cache:
+        return _handle_combinatory_strategy(query, None, strategy)
+
+    # Redis object names
+    CURRENT_MAX_ID = f'{g.active_organisation}_{object_model.__table__.name}_max_id'
```

Comment:
thoughts on using `f'org{g.active_organisation.id}...` instead? not really a big deal, but results in a slightly cleaner key